book/build-service.md

```diff
@@ -0,0 +1,142 @@
+  - The prerequisite is to have installed GitHub App which is used by the Build Server in the user's repository.
+
+Default mode:
+1. Wait until the component has `.status.devfile` set
+1. Get data from `.status.devfile` and use them for creating PipelineRun
+1. Set annotation on the component - `appstudio.openshift.io/component-initial-build: processed`
+
+Delete annotation `appstudio.openshift.io/component-initial-build` on Component to retrigger PipelineRun in Default mode.
+
+For the selection of Custom mode the Component CR must have set annotation `appstudio.openshift.io/pac-provision` to `request`.
+
+Custom Mode:
+1. Wait until the component has `.status.devfile` set
+1. Get data from `.status.devfile`, use them for creating `.tekton` folder in the user's repository and create Repository CR so that Pipelines-as-Code is configured to monitor the user's repository.
+1. Change the value of `appstudio.openshift.io/pac-provision` annotation to `done` in case the previous step was successful, in case of an issue the annotation is set to `error` and annotation `appstudio.openshift.io/pac-provision-error` with more details about the error is added.
+
+#### PipelineRun selection
+
+The Build Service owns [BuildPipelineSelector CRD](https://redhat-appstudio.github.io/book/ref/build-service.html#buildpipelineselector) which defines which PipelineRun to select for the Component CR. By default global BuildPipelineSelector `build-pipeline-selector` in `build-service` namespace is used. BuildPipelineSelector CR contains selectors, the first matching selector is used for the Component CR. The list of selectors can be extended by creating BuildPipelineSelector CR with the same name as the Application CR in the user's namespace. This will ensure that it is applied to Component CRs under the corresponding Application CR. The list of selectors can be also extended for the whole user namespace by creating BuildPipelineSelector CR named `build-pipeline-selector` in the user's namespace.
+
+Example of BuildPipelineSelector:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: BuildPipelineSelector
+metadata:
+  name: build-pipeline-selector
+  namespace: my-tenant
+spec:
+  selectors:
+  - name: my-component
+    pipelineRef:
+      bundle: quay.io/my-user/my-bundle:v1.0
+      name: my-bundle
+    pipelineParams:
+    - name: test
+      value: fbc
+    when:
+      language: java
+      componentName: my-component
+      projectType: springboot
+  - name: Docker build
+    pipelineRef:
+      bundle: quay.io/redhat-appstudio-tekton-catalog/pipeline-docker-build:3649b8ca452e7f97e016310fccdfb815e4c0aa7e
+      name: docker-build
+    when:
+      dockerfile: true
+  - name: NodeJS
+    pipelineRef:
+      bundle: quay.io/redhat-appstudio-tekton-catalog/pipeline-nodejs-builder:3649b8ca452e7f97e016310fccdfb815e4c0aa7e
+      name: nodejs-builder
+    when:
+      language: nodejs,node
+```
```

Comment:
Please add a fallback case without `when` statement.
Also, an example of matching label (maybe with OR condition for values) would be useful as well.

Reply:
added